Modules/Core/TestKernel/test/itkTestingComparisonImageFilterGTest.cxx

```diff
@@ -0,0 +1,106 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkTestingComparisonImageFilter.h"
+#include "itkImage.h"
+#include "itkVectorImage.h"
+
+#include "itkGTest.h"
+
+
+// A Google test to check the class's run time type information
+TEST(itkTestingComparisonImageFilterTest, CheckRunTimeTypeInformation)
+{
+  // Create an image
+  using ImageType = itk::Image<unsigned char, 2>;
+  auto filter = itk::Testing::ComparisonImageFilter<ImageType, ImageType>::New();
+
+  // Test the run-time type information
+  EXPECT_STREQ(filter->GetNameOfClass(), "ComparisonImageFilter");
+  EXPECT_NO_THROW(filter->Print(std::cout));
+}
+
+// Test two 16x16 images of all zeros, and check all statistics are zero
+TEST(itkTestingComparisonImageFilterTest, TestZeroImages)
+{
+  // Create two 16x16 images of all zeros
+  using ImageType = itk::Image<unsigned short, 2>;
+  auto                image1 = ImageType::New();
+  auto                image2 = ImageType::New();
+  ImageType::SizeType size = { { 16, 16 } };
+  image1->SetRegions(size);
+  image2->SetRegions(size);
+  image1->AllocateInitialized();
+  image2->AllocateInitialized();
+  image1->FillBuffer(0);
+  image2->FillBuffer(0);
```

Comment:
@blowekamp These two `FillBuffer(0)` calls could have been removed, of course! Sorry I did not notice them while reviewing 🤷